pkgs/by-name/br/broadcast-box/package.nix

```
@@ -0,0 +1,69 @@
+{
+  lib,
+  nixosTests,
+  fetchFromGitHub,
+  buildNpmPackage,
+  buildGoModule,
+}:
+let
+  name = "broadcast-box";
+  version = "2024-07-15";
+
+  src = fetchFromGitHub {
+    repo = "broadcast-box";
+    owner = "Glimesh";
+    rev = "323a3f87adff0fcecd7292eae960d666b6f52787";
+    hash = "sha256-uKmh0JtzIaO/yGicgcoXV2lRDfjHeJ44qnQ5jFUb8UM=";
+  };
+
+  frontend = buildNpmPackage {
+    inherit version;
+    pname = "${name}-web";
+    src = "${src}/web";
+    npmDepsHash = "sha256-tCKpnvrEGT/y/XaWdfNOTPow/DJCmbZKlv1acLwd3Ew=";
+    preBuild = ''
+      # The REACT_APP_API_PATH environment variable is needed
+      cp "${src}/.env.production" ../
+    '';
+    installPhase = ''
+      mkdir -p $out
+      cp -r build $out
+    '';
+  };
+in
+buildGoModule {
+  inherit version src frontend;
+  pname = name;
+  vendorHash = "sha256-nYsSFo64WG+UyPDVBIgC7Lc0J4wo6334PNAVPvQQSfY=";
+  proxyVendor = true; # fixes darwin/linux hash mismatch
+
+  patches = [ ./allow-no-env.patch ];
+  postPatch = ''
+    substituteInPlace main.go \
+      --replace-fail './web/build' '${placeholder "out"}/share'
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/share
+    cp -r ${frontend}/build/* $out/share
```

I would prefer this to be like this:
```suggestion
    cp -r $frontend/build/* $out/share
```

Note that `${}` is evaluated by Nix so the final value of `installPhase` would contain a static reference to the store path of the `frontend` derivation. If you wish to use `${}` as part of shell syntax, you need to escape the $ with two single quotes like this:

```suggestion
    cp -r ''${frontend}/build/* $out/share
```

This allows users to override the frontend package if they ever wish to do so, like this:

```nix
pkgs.broadcast-box.overrideAttrs (prevAttrs: {
  frontend = prevAttrs.frontend.override {...};
});
```